### ISSUE TITLE
Ubuntu 20.04 obs-ndi.so libQt6Widgets.so.6: cannot open shared object file error

### ISSUE THREAD

**Issue by gyalokai** [obs-ndi/obs-ndi#868]
Hi,

I tried to install version 4.11.1 as described on an intel build PC (vga: nVidia GTX 1660 super) Ubuntu 20.04, with latest OBS (v29.0.2) but the obs-ndi plugin can not be loaded.

I found in the log this:

os_dlopen(/usr//lib/x86_64-linux-gnu/obs-plugins/obs-ndi.so->/usr//lib/x86_64-linux-gnu/obs-plugins/obs-ndi.so): libQt6Widgets.so.6: cannot open shared object file: No such file or directory

Module '/usr//lib/x86_64-linux-gnu/obs-plugins/obs-ndi.so' not loaded

What could be the problem? I could not find anything in connection with this file "libQt6Widgets.so.6" elsewhere.

EDIT: 

git clone https://github.com/obs-ndi/obs-ndi.git
cd obs-ndi
.github/scripts/build-linux.sh

Error messages I get:
E: Unable to locate package qt6-base-dev
E: Unable to locate package libqt6svg6-dev
E: Unable to locate package qt6-base-private-dev
    X script execution error
    Callstack:
    /root/install/ndi/obs-ndi/.github/scripts/utils.zsh/setup_linux:55
    /root/install/ndi/obs-ndi/.github/scripts/build-linux.zsh:168
    /root/install/ndi/obs-ndi/.github/scripts/build-linux.zsh:265

What do you need for debugging?

Thank you any help in advance!

**Comment by rcooke** [comment]
I'm dealing with this problem myself.  I had a problem with my capture card driver under Ubuntu 22, so I rebuilt the machine with Ubuntu 20 LTS.  I installed latest OBS (V28) and OBS-NDI (4.11.1) plugin binary. It appears the plugin was compiled with QT6 because my error was it could not find "libQt6Widgets.so".  This lead me to a solution that should work for you as well: [here](https://askubuntu.com/a/1425079).

The steps to fix involve adding a back-port repo to your system, then installing the QT6 packages:

```
sudo add-apt-repository ppa:okirby/qt6-backports
sudo apt update
sudo apt install qt6-base-dev libqt6svg6-dev qt6-base-private-dev
```

While this worked for me, I ran into another problem with the pre-made binaries.  Now I get an error that my GNU C-lib is the wrong version.  Again, a difference between Ubuntu 20 and 22.  I have GLIBC_2.31 but the plugin binary wants GLIBC_2.34

So next I will try what you tried: compile from source.  With the QT6 libs installed, this should work.  Give a minute to find out...

**Comment by gyalokai** [comment]
Hi Richard,

thank you very much for responding to my post and I also apologize, but I decided to let go of ubuntu version 20.04. 

Since 20.04 is only supported until April 2025 (less than 2 years to go) and the server was a relatively young installation, I decided to upgrade to 22.04 instead.

From then on, everything went smoothly, I've been using it ever since, OBS 29.1.1 with NDI 5.x is running perfectly.
